packages/generator/src/sdk-metadata/generation-and-usage.spec.ts

```diff
@@ -2,34 +2,43 @@ import { resolve } from 'path';
 import { checkUrlExists } from '@sap-cloud-sdk/test-util';
 import { writeFile, readFile, removeSync } from 'fs-extra';
 import execa = require('execa');
+import { VdmServiceMetadata } from '../vdm-types';
 import {
   getApiSpecificUsage,
-  getGenerationDocumentation,
+  getGenerationAndUsage,
+  getGenerationSteps,
   getGenericUsage,
   linkGenerationDocumentaion
 } from './generation-and-usage';
 
 describe('generation-and-usage', () => {
+  const service = {
+    npmPackageName: '@sap/dummy-package',
+    entities: [{ className: 'DummyClass' }]
+  } as VdmServiceMetadata;
+
   it('creates generic usage example', async () => {
     await expect(getGenericUsage()).resolves.toMatchSnapshot();
   });
 
   it('creates api specific usage for entity', async () => {
-    await expect(
-      getApiSpecificUsage({
-        npmPackageName: '@sap/dummy-package',
-        entities: [{ className: 'DummyClass' }]
-      } as any)
-    ).resolves.toMatchSnapshot();
+    await expect(getApiSpecificUsage(service)).resolves.toMatchSnapshot();
+  });
+
+  it('[E2E] gives a working generator repository link', async () => {
+    const generationAndUsage = await getGenerationAndUsage(service);
+    checkUrlExists(generationAndUsage.generatorRepositoryLink);
```

Comment:
Do we really need to check that the URL exists? This also applies to the already existing tests...

Reply:
I think we should do that. Imagine we change some path in the docs, who remembers to update the sdk-metatdata?

Reply:
Hmm, true.
Somehow I am not a fan of putting this together with the unit tests. I would rather expect this to be a separate .spec in the integration tests.
If no code is changed, I would expect no (unit) tests to fail, which could happen here. It also takes a little longer than other tests I assume.

(I noticed that we have the [E2E] prefix here, which I am also not a fan of. I think if it is an E2E test, it should be even more explicit than just the description).

If you agree, maybe move it, but I won't insist, especially because we already had some tests in this direction. As long as we don't have issues, we can look at this as a matter of taste ;)

Reply:
Maybe one additional point: I would also expect the unit tests to be able to run offline without external factors.

Reply:
moved it to E2E test see next PR.